securesystemslib/gpg/constants.py

```diff
@@ -15,15 +15,17 @@
   aggregates all the constant definitions and lookup structures for signature
   handling
 """
+import functools
 import logging
 import os
 
 from securesystemslib import process
 
 log = logging.getLogger(__name__)
 
-
-def is_available_gnupg(gnupg):
+@functools.lru_cache(maxsize=3)
```

Comment:
I have to put a maxsize for py3.7, which is in the tox matrix. Otherwise would use the default.

I picked 3 because we should only call it three times under normal circumstances (`gpg`, `gpg2`, `$GNUPG`). That's perhaps a little bit too clever but I had a hard time justifying other numbers.